datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -128,68 +129,32 @@ struct SkipAggregationProbe {
     /// Number of processed input rows (updated during probing)
     input_rows: usize,
-    /// Number of total group values for `input_rows` (updated during probing)
-    num_groups: usize,
 
-    /// Flag indicating further data aggregation may be skipped (decision made
-    /// when probing complete)
-    should_skip: bool,
-    /// Flag indicating further updates of `SkipAggregationProbe` state won't
-    /// make any effect (set either while probing or on probing completion)
-    is_locked: bool,
-
-    /// Number of rows where state was output without aggregation.
-    ///
-    /// * If 0, all input rows were aggregated (should_skip was always false)
-    ///
-    /// * if greater than zero, the number of rows which were output directly
-    ///   without aggregation
-    skipped_aggregation_rows: metrics::Count,
+    /// Number of unique hash, which represents the cardinality of the group values
+    unique_hashes_count: HashSet<u64>,
 }
 
 impl SkipAggregationProbe {
-    fn new(
-        probe_rows_threshold: usize,
-        probe_ratio_threshold: f64,
-        skipped_aggregation_rows: metrics::Count,
-    ) -> Self {
+    fn new(probe_ratio_threshold: f64) -> Self {
         Self {
             input_rows: 0,
-            num_groups: 0,
-            probe_rows_threshold,
             probe_ratio_threshold,
-            should_skip: false,
-            is_locked: false,
-            skipped_aggregation_rows,
+            unique_hashes_count: Default::default(),
         }
     }
 
     /// Updates `SkipAggregationProbe` state:
-    /// - increments the number of input rows
-    /// - replaces the number of groups with the new value
-    /// - on `probe_rows_threshold` exceeded calculates
-    ///   aggregation ratio and sets `should_skip` flag
-    /// - if `should_skip` is set, locks further state updates
-    fn update_state(&mut self, input_rows: usize, num_groups: usize) {
-        if self.is_locked {
-            return;
+    /// Insert hashes to the HashSet, if the number of
+    /// unique hashes to the total rows exceed the
+    /// threshold, return true to indicates `should skip aggregation`
+    fn update_state(&mut self, batch_hashes: &[u64]) -> bool {
+        for target_hash in batch_hashes.iter() {
+            self.unique_hashes_count.insert(*target_hash);
         }
-        self.input_rows += input_rows;
-        self.num_groups = num_groups;
-        if self.input_rows >= self.probe_rows_threshold {
-            self.should_skip = self.num_groups as f64 / self.input_rows as f64
-                >= self.probe_ratio_threshold;
-            self.is_locked = true;
-        }
-    }
+        self.input_rows += batch_hashes.len();
 
-    fn should_skip(&self) -> bool {
-        self.should_skip
-    }
-
-    /// Record the number of rows that were output directly without aggregation
-    fn record_skipped(&mut self, batch: &RecordBatch) {
-        self.skipped_aggregation_rows.add(batch.num_rows());
```

Comment:
Not sure if this metric is not required -- it still may be helpful for debugging purposes or just providing information about how aggregation has been executed to end user (via `explain analyze`)